Framework/SwiftOSC/Communication/OSCServer.swift

```diff
@@ -11,36 +16,70 @@ extension OSCServerDelegate {
     public func didReceive(_ message: OSCMessage){}
 }
 
+
+
+
+
+/// Used to receive OSC messages from another source, which are sent to the delegate.
 open class OSCServer {
+    
+    /// Returns the current address, or sets the address messages will be sent to.
     open var address: String {
         didSet {
             _ = server.close()
             server = UDPServer(addr: self.address, port:self.port)
         }
     }
+    
+    /// Returns the current port, or sets the port messages will be sent to.
     open var port: Int {
         didSet {
             _ = server.close()
             server = UDPServer(addr: self.address, port:self.port)
         }
     }
+    
+    /// Callbacks are send to this instance.
     open var delegate: OSCServerDelegate?
-    open var running = false
+    
+    /// Returns the state of the server if its enabled or not.
+    open fileprivate(set) var running = false
+    
     var server: UDPServer
     
+    
+    /**
+     Creates a new OSCServer
+     
+     - Parameters:
+       - address: Could be an IP address, or just "localhost" for sending internally in your computer. For most cases, just leave this with a blank String
+       - port: Must be a vaild port that's not being used.
+     
+     ~~~
+     // An example of usage
+     let server = OSCServer(address: "", port: 8080)
+     ~~~
+    */
     public init(address: String, port: Int){
         self.address = address
         self.port = port
         self.server = UDPServer(addr: self.address, port:self.port)
         run()
     }
     
-    open func start(){
+    
+    /// Allows messages to be received
+    open func start() {
         running = true
     }
+    
+    
+    /// Stops messages from being received. Does not free up the port.
```

Comment:
This is a good point. Should I change this to free up the port?

Reply:
If I wanted to free up the port, I would deinit the instance. I think this library is meant to be as simple as possible, which is great. I think by not freeing the port, you're protecting the user who may not know a lot about networking.

Reply:
Okay we'll leave it as is.